src/index.js

```diff
@@ -205,11 +206,17 @@ export default class DataLoader<K, V> {
 // In order to avoid the DataLoader dispatch Job occuring before "PromiseJobs",
 // A Promise Job is created with the sole purpose of enqueuing a global Job,
 // ensuring that it always occurs after "PromiseJobs" ends.
-function enqueuePostPromiseJob(fn) {
+function enqueuePostPromiseJob(fn, _environment: Environment) {
   if (!resolvedPromise) {
     resolvedPromise = Promise.resolve();
   }
-  resolvedPromise.then(() => process.nextTick(fn));
+  resolvedPromise.then(() => {
+    if (_environment.isNode) {
+      process.nextTick(fn);
+    } else {
+      setTimeout(fn);
+    }
```

Comment:
Personally, I'd simply inline the check for node here. The overhead of a complete `Environment` class seems rather heavy to me. Most runtimes I can think of at the moment support `setTimeout` so we'd only have to check for node.

Something like this would suffice in my opinion:

```js
resolvedPromise.then(() => {
  if (process !== undefined && process.nextTick !== undefined) {
    process.nextTick(fn);
  } else {
    setTimeout(fn);
  }
});
```

(Note though that I'm not a contributor to dataloader. The core members probably have their reasons)

Reply:
Yeah That makes sense to me.  I initially thought to do it that way, but wasn't quite sure how to get 100% test coverage. Do you know how i could get this logic `process !== undefined && process.nextTick !== undefined` to return false in the tests?